log/format.go

```diff
@@ -30,70 +30,42 @@ type TerminalStringer interface {
 	TerminalString() string
 }
 
+// format formats the log record for terminal output
 func (h *TerminalHandler) format(buf []byte, r slog.Record, usecolor bool) []byte {
 	msg := escapeMessage(r.Message)
-	var color = ""
-	if usecolor {
-		switch r.Level {
-		case LevelCrit:
-			color = "\x1b[35m"
-		case slog.LevelError:
-			color = "\x1b[31m"
-		case slog.LevelWarn:
-			color = "\x1b[33m"
-		case slog.LevelInfo:
-			color = "\x1b[32m"
-		case slog.LevelDebug:
-			color = "\x1b[36m"
-		case LevelTrace:
-			color = "\x1b[34m"
-		}
-	}
+	color := determineColor(usecolor, r.Level)
+
 	if buf == nil {
 		buf = make([]byte, 0, 30+termMsgJust)
 	}
 	b := bytes.NewBuffer(buf)
 
-	if color != "" { // Start color
-		b.WriteString(color)
-		b.WriteString(LevelAlignedString(r.Level))
-		b.WriteString("\x1b[0m")
-	} else {
-		b.WriteString(LevelAlignedString(r.Level))
-	}
+	b.WriteString(color)
+	b.WriteString(LevelAlignedString(r.Level))
+	b.WriteString("\x1b[0m")
 	b.WriteString("[")
 	writeTimeTermFormat(b, r.Time)
 	b.WriteString("] ")
 	b.WriteString(msg)
 
-	// try to justify the log output for short messages
-	//length := utf8.RuneCountInString(msg)
 	length := len(msg)
 	if (r.NumAttrs()+len(h.attrs)) > 0 && length < termMsgJust {
 		b.Write(spaces[:termMsgJust-length])
 	}
-	// print the attributes
 	h.formatAttributes(b, r, color)
 
 	return b.Bytes()
 }
 
+// formatAttributes formats the attributes of the log record
 func (h *TerminalHandler) formatAttributes(buf *bytes.Buffer, r slog.Record, color string) {
 	writeAttr := func(attr slog.Attr, first, last bool) {
 		buf.WriteByte(' ')
-
-		if color != "" {
-			buf.WriteString(color)
-			buf.Write(appendEscapeString(buf.AvailableBuffer(), attr.Key))
-			buf.WriteString("\x1b[0m=")
-		} else {
-			buf.Write(appendEscapeString(buf.AvailableBuffer(), attr.Key))
-			buf.WriteByte('=')
-		}
+		buf.WriteString(color)
+		buf.Write(appendEscapeString(buf.AvailableBuffer(), attr.Key))
+		buf.WriteString("\x1b[0m=")
```

Comment:
Pretty sure that this is wrong, previously we wrote `=` when color was "", now its "\x1b[0m="